src/js/ripple/transaction.js

```diff
@@ -375,21 +393,19 @@ Transaction.prototype.complete = function() {
   }
 
   // Try to auto-fill the secret
-  if (!this._secret && !(this._secret = this.getSecret())) {
+  if (!this._keyPair && !(this._keyPair = this.getKeyPair()) && 
+      !this._secret && !(this._secret = this.getSecret())) {
     this.emit('error', new RippleError('tejSecretUnknown', 'Missing secret'));
     return false;
   }
 
   if (typeof this.tx_json.SigningPubKey === 'undefined') {
-    try {
-      var seed = Seed.from_json(this._secret);
-      var key  = seed.get_key(this.tx_json.Account);
-      this.tx_json.SigningPubKey = key.to_hex_pub();
-    } catch(e) {
-      this.emit('error', new RippleError(
-        'tejSecretInvalid', 'Invalid secret'));
+    var key = this._keyPair;
+    if (!key && this.remote && !(key = this.remote.generateKeyPair(this._secret))) {
```

Comment:
how about adding back a generateKeyPair() in Transaction.js?

https://github.com/ripplerm/ripple-lib/commit/8ddfb8898d263c2db08a60553147ba8344f049e6